manager-dashboard/app/views/NewProject/index.tsx

```diff
@@ -271,9 +274,69 @@ function NewProject(props: Props) {
                 visibility,
                 filter,
                 filterText,
+                projectTopic,
                 ...valuesToCopy
             } = finalValues;
 
+            try {
+                const db = getDatabase();
+                const projectRef = databaseRef(db, 'v2/projects/');
+                const projectTopicKey = projectTopic?.toLowerCase() as string;
+
+                const prevProjectNameQuery = query(
+                    projectRef,
+                    orderByChild('projectTopicKey'),
+                    equalTo(projectTopicKey),
+                );
+
+                const snapshot = await getValueFromFirebase(prevProjectNameQuery);
+
+                if (snapshot.exists()) {
+                    setError((prevErr) => ({
+                        ...getErrorObject(prevErr),
+                        [nonFieldError]: 'A project with this name already exists, please use a different project name (Please note that the name comparision is not case sensitive)',
+                        projectTopic: 'A project with this name already exists, please use a different project name (Please note that the name comparision is not case sensitive)',
+                    }));
+                    setProjectSubmissionStatus(undefined);
+                    return;
+                }
+
+                const newProjectRef = await pushToDatabase(projectRef);
+                const newKey = newProjectRef.key;
+
+                if (!mountedRef.current) {
+                    return;
+                }
+
+                if (!newKey) {
+                    setError((err) => ({
+                        ...getErrorObject(err),
+                        [nonFieldError]: 'Failed to push new key for the project',
+                    }));
+                    setProjectSubmissionStatus('failed');
+                    return;
+                }
+
+                const uploadData = {
+                    ...finalValues,
+                    projectTopicKey,
+                    createdAt: (new Date()).getTime(),
+                };
+
+                const putProjectRef = databaseRef(db, `v2/projects/${newKey}`);
+                await setToDatabase(putProjectRef, uploadData);
```

Comment:
What is the difference between v2/projects and v2/projectDrafts? We are also creating projectDrafts later